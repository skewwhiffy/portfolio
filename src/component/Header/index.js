import React  from 'react'
import Navigation from '../Navigation'

const Header = props => {
  const { choices, currentChoice, setCurrentChoice } = props
  return (
    <div className="header">
      <h1>Kenny Hung</h1>
      <Navigation
        choices={choices}
        currentChoice={currentChoice}
        setCurrentChoice={setCurrentChoice}
      />
    </div>
  )
}

export default Header
