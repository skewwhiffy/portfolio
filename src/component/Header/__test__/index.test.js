import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Header from '..';

afterEach(cleanup)

const currentChoice = {
  name: 'choice',
  displayName: 'Dummy Choice For Testing'
}
const anotherChoice = {
  name: 'anotherChoice',
  displayName: 'Another Choice For Testing'
}
const choices = [currentChoice, anotherChoice]

describe('Header component', () => {
  it('matches snapshot DOM node structure', () => {
    const { asFragment } = render(
      <Header
        choices={choices}
        currentChoice={currentChoice}
        setCurrentChoice={jest.fn()}
      />
    )

    expect(asFragment()).toMatchSnapshot();
  });
})