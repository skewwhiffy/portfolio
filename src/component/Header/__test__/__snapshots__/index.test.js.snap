// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div
    class="header"
  >
    <h1>
      Kenny Hung
    </h1>
    <div
      class="navigation"
    >
      <ul>
        <li
          class="selected"
        >
          <span
            class="selection"
          >
            Dummy Choice For Testing
          </span>
        </li>
        <li
          class=""
        >
          <span
            class="selection"
          >
            Another Choice For Testing
          </span>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
